enhancements/cli-in-cluster-management.md

```diff
@@ -0,0 +1,219 @@
+    * An openshift-krew-index with supported artifacts could hold information about CLIs supported by OpenShift. However, with `krew index add https://someother-third-party-index` 
+    we won't limit cluster-admins from adding their own index with whatever plugins they want.  Only the openshift-krew-index will be supported.
+
+Initial implementation will be with Option 3. Future implementation can combine 3 and 4. Each OpenShift CLI will publish artifacts, for example, [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/)
+and also create an image similar to this but that also includes sha256sum files: [example cli-artifacts image](https://github.com/openshift/oc/blob/master/images/cli-artifacts/Dockerfile.rhel).  CLI artifacts are packaged in images.
+
+Users will continue to have the option to download binaries from either the central location where artifacts are published or from the console.  
+This proposal adds the mechanism to install from images built from the same artifacts using `oc`. All options will offer the same binaries, 
+ensured through sha256 signatures.  
+
+For the initial implementation, supported CLIs will create ClusterCLI custom resources.  The CLI will be installed to a user's home directory with
+the same logic as `oc image extract` and from each CLI's artifacts image (see oc, cli-artifacts image for reference).  
+Through mirrored images and ImageContentSourcePolicy, CLIs will be available to disconnected environments. Currently, the console provides
+oc binaries by running a webserver pod in the console namespace with a route.  The cli-manager will take over running this webserver, this will
+be included in the CRD controller that will create the core ClusterCLI custom resources (oc, odo, and helm, the 3 CLIs currently available 
+through the consoleclidownloads mechansism).
+
+### Non-Goals
+
+* `oc clustercli` will not build or serve the binaries.  It will know where to find them.
+* This proposal is not concerned with _which_ binaries will be managed.  This proposal is meant to determine the mechanism only.  Consumers and publishers are clients of the mechanism. 
+* `oc clustercli` will not create or update the ClusterCLI Custom Resources, that will be managed by individual CLIs. However, the CLIs that are 
+currently available as `consoleclidownloads`, `oc, odo, and helm`, will be included in the reference implementation of this enhancement.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a user, I want a CLI manager for various CLIs available for OpenShift and related services so that I can discover, install and list them
+The user will invoke the following commands:
+
+* `oc clustercli list` will access the connected OpenShift cluster, if connected,  and will retrieve information about available CLIs (clusterCLI CRs)
+* `oc clustercli install odo` will install `odo` to a user's home directory, as a standalone binary.
```

Comment:
That should be great to have a command such as `oc clustercli install kubetools` able to install the recommended tools: helm, kubectl, oc, odo, k9s, ... - WDYT ?